lib/bolt/transport/local_windows.rb

```diff
@@ -98,14 +70,12 @@ def execute(*command, options)
         result_output
       end
 
-      def upload(target, source, destination, options = {})
-        self.class.validate(options)
```

Comment:
None of the other transports validate options in `#upload` or the `#run_*` methods. It looks like this is only here so we can issue a warning when the transport receives the `run-as` option, either through config or as an option in a plan function.

Issuing a warning from the transport config class is a little tricky, since the Logger is configured after the class. We could add a list of warnings in the transport config that are printed later (similar to what we do for the option parser).

Adding a specific warning for instances when the `_run_as` option is used in a plan function should be easy enough.